sparklightautoml/dataset/base.py

```diff
@@ -69,6 +78,32 @@ def _set_col(data: Any, k: int, val: Any):
 
     ID_COLUMN = "_id"
 
+    @classmethod
+    def load(cls,
+             path: str,
+             file_format: str = 'parquet',
+             file_format_options: Optional[Dict[str, Any]] = None,
+             persistence_manager: Optional['PersistenceManager'] = None,
+             partitions_num: Optional[int] = None) -> 'SparkDataset':
+        metadata_file_path = os.path.join(path, f"metadata.{file_format}")
+        file_path = os.path.join(path, f"data.{file_format}")
+        file_format_options = file_format_options or dict()
+        spark = get_current_session()
+
+        # reading metadata
+        metadata_df = spark.read.format(file_format).options(**file_format_options).load(metadata_file_path)
+        metadata = pickle.loads(metadata_df.select('metadata').first().asDict()['metadata'])
```

Comment:
please consider removing pickle as metadata format. even json could be more safe without huge performance hit

Reply:
fixed.
But json solution may be a problem if SparkTask start using custom metric and loss functions provided by user at some point.